revision.c

```diff
@@ -624,11 +626,119 @@ static void file_change(struct diff_options *options,
 	options->flags.has_changes = 1;
 }
 
+static int bloom_filter_atexit_registered;
+static unsigned int count_bloom_filter_maybe;
+static unsigned int count_bloom_filter_definitely_not;
+static unsigned int count_bloom_filter_false_positive;
+static unsigned int count_bloom_filter_not_present;
+static unsigned int count_bloom_filter_length_zero; 
```

Comment:
I wonder if these should be grouped into a static struct?

```
static struct {
    unsigned int count_maybe;
...
} bloom_filter_data;
```

We may also want to `memset()` to zero during `prepare_to_use_bloom_filter()`.

Reply:
@derrickstolee I will do that in a follow up or something. Don't feel very strongly for or against.